### PR TITLE
feat: update neutronjs to sdk47 #ntrn-146

### PR DESCRIPTION
**Pull request by NeverHappened**
https://hadronlabs.atlassian.net/browse/NTRN-147

What's done:
1. Uses buf to generate protobuf generated files
2. Moved all newer sdk47 branch code for sdk47 into neutronjs
3. Removed all helper's code from integration tests

Integration tests: https://github.com/neutron-org/neutron-integration-tests/pull/244
NeutronJS: https://github.com/neutron-org/neutronjsplus/pull/11

### CODE REVIEW

File: src/common_localcosmosnet.ts [hunk 1]
```diff
@@ -187,16 +187,17 @@ export class TestStateLocalCosmosTestNet {
...
       try {
-        const sequence = await cm.chain.getSeq(cm.wallet.address);
+        const sequence = await cm.chain.getSeq(cm.wallet.address.toString());
         res = await cm.msgSend(
           to.toString(),
           { amount, denom },
           fee,
           sequence,
-          rest.tx.BroadcastTxMode.Block,
+          cosmosclient.rest.tx.BroadcastTxMode.Async,
```
Comment (pr0n00gler) [inline src/common_localcosmosnet.ts:196]: Why not `Sync` mode?

File: src/common_localcosmosnet.ts [hunk 2]
```diff
@@ -187,16 +187,17 @@ export class TestStateLocalCosmosTestNet {
...
         );
         break;
       } catch (e) {
+        console.log(e);
```
Comment (pr0n00gler) [inline src/common_localcosmosnet.ts:200]: Redundant log?

**Review by pr0n00gler — COMMENTED** [review]
What's the reason of moving files from `src/helpers` folder into `src`?

**Comment by NeverHappened** [comment]
https://github.com/neutron-org/neutronjsplus/pull/11#pullrequestreview-1755071868
reason is that it's redundant directory, since all the library code was in "helpers".
I think it's the old legacy of the code being in neutron-integration-tests, where it made sense.
So I think in the future (refactoring) we can move files under some meaningful structure, but `helpers` is not it.

File: src/common_localcosmosnet.ts (hunk 1 above)
Comment (NeverHappened) [reply src/common_localcosmosnet.ts:196]: Guess no reason, will change to Sync

**Comment by NeverHappened** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7046159446

**Review by pr0n00gler — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

File: gen-proto.sh [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
+git clone https://github.com/neutron-org/cosmos-sdk $REPOS/cosmos-sdk --branch v0.47.6-neutron --depth 1
+npx buf generate $REPOS/cosmos-sdk/proto --template $TEMPLATE --output $OUT/cosmos_sdk --include-imports
+
+# ===== ibc-go ================
+git clone https://github.com/cosmos/ibc-go $REPOS/ibc-go --branch v7.3.1 --depth 1
+npx buf generate $REPOS/ibc-go/proto --template $TEMPLATE --output $OUT/ibc_go --include-imports
+
+# note: commented because we don't have pob for now
+# ===== block-sdk (pob) =======
+git clone git@github.com:skip-mev/block-sdk.git $REPOS/block-sdk --branch v1.1.0 --depth 1
```
Comment (pr0n00gler) [inline gen-proto.sh:31]: We don't have block-sdk now

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Please add info into Readme how to generate protobuf

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@neutron-org/neutronjsplus",
-  "version": "0.0.16",
+  "version": "0.1.0-rc4",
```
Comment (sotnikov-s) [inline package.json:3]: I think we should have a mapping neutron version <-> neutronjsplus version somewhere, for example in the repo readme. let's add it? at least starting with this version: neutron v2.0.0 <-> neutronjsplus 0.1.0

File: gen-proto.sh (hunk 3 above)
Comment (pr0n00gler) [reply gen-proto.sh:31]: Decided to leave, since we'll include block-sdk in a month anyway

**Review by pr0n00gler — DISMISSED** [review]

**Pull request merged**